2019Labs/CustomSecurityContent/documentation/lab3_oval.adoc

```diff
@@ -0,0 +1,487 @@
+* ... how to test OVAL checks.
+* ... how to use tests and remediations to safely and gradually improve an OVAL check.
+
+
+== What have we done for you
+
+* We have cloned the `ComplianceAsCode` repository to the `content` directory.
+* We have installed dependencies that are required for the content build using `dnf install`:
+** generic build utilities: `cmake`, `make`,
+** utilities for generating SCAP content: `openscap-utils`, and finally
+** python dependencies for putting content together: `python3-pyyaml`, `python3-jinja2`.
+
+* We have set up the `podman` container by building one using the `Dockerfiles/test_suite-rhel` files.
+`root` 's  SSH keys are authorized by the container's root user.
+* We have modified the OVAL check for the `accounts_tmout`, so we can improve it.
+
+
+== Hands-on
+
+
+=== Before you start
+
+* Open a terminal window by clicking `Activities` in the upper left corner, and then click on the terminal icon.
+* Make sure that you are in the project root when you start - run
++
+.....
+cd /home/lab-user/labs/lab3_oval
+.....
++
+in the terminal.
+
+* For best experience, open one more terminal window and also navigate to the project root in that one as well.
+
+
+=== The current state
+
+Make sure that you are in the `content` directory - we will build the content for the RHEL8 product.
+Execute the `./build_product rhel8` command.
+
+When the build finishes, let's take a look at the RHEL8 OSPP profile:
+
+----
+firefox build/guides/ssg-rhel8-guide-ospp.html
+----
+
+In this track, we will focus on the `accounts_tmout` rule.
+In order to find the rule entry it in the guide, press `Ctrl+F` and search for the `Set Interactive Session Timeout` string, which is the rule title.
+
+The description says:
+
+____
+
+Setting the TMOUT option in /etc/profile ensures that all user sessions will terminate based on inactivity. The TMOUT setting in /etc/profile should read as follows:
+
+TMOUT=600
+____
+
+
+When dealing with the rule check, we will focus on those two aspects:
+
+- What if the timeout is set to 100?
+Is it more or less secure?
++
+Having shorter time interval between inactivity and logout is more bothersome for the user, but it is a stricter requirement.
+Therefore, we would like to make sure that the if a rule that requires `TMOUT=600` also allows `TMOUT=100`.
+
+- The rule description of having the `TMOUT=...` statement in a config file is accurate, but guides on the Internet often recommend to have `export TMOUT=...` there.
+We want to make sure that the rule's check will allow both forms - with and without `export`, although the `export` keyword is not required.
+
+
+==== Bash remediation
+
+Let's examine the bash remediation - as you know from the introductory lab, you can either
+- scroll down the rule entry in the HTML guide and click the `(show)` link next to the `Remediation Shell Script` label, or
+- open the `linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh` file.
```

Comment:
You give them 2 alternatives, but they aren't the same. The entry in HTML guide contains variable substituted by its values, but the snippet you show below contains  `var_accounts_tmou` unexpanded.